include/zos-setlibpath.h

```diff
@@ -29,9 +29,12 @@ class __Z_EXPORT __setlibpath {
     while ((token = w_getpsent(token, &buf, sizeof(buf))) > 0) {
```

Comment:
Move the body of this ctor into the library code.  It is much too big to put in the header and has lots of library calls.  You'll be able to clean up the #include's in this header too.

Reply:
This header is used (currently by Node.js) **before** libzoslib.so has been loaded - to find that .so and set LIBPATH to its path; so this code can't be part of the library.

Reply:
Wouldn't it be useful for other apps?, where in general you have x/bin/app and x/lib/lib.so, so app can be run without user having to set LIBPATH. I can remove lib.target so the class applies to the more general cases.

Reply:
Let me know if the above would do, or if there's other generally used lib paths to include here so it's useful for others, or to just remove the file.

Reply:
The general solution would be to enable the OS to save LIBPATH & STEPLIB into the executable.  If we were to keep this in zoslib I'd say:
1. package this code into a separate .a library.  So still move the code out of the header and into a .o file in the archive.
2. rename the class to remove the double underscores.  
3. add a template parameter for template of what you want the LIBPATH to look like. eg `"%EXEC%/lib.target:%EXEC%/lib:

I'd leave the code out of zoslib since it is tailored towards node.js.  Most people use STEPLIB over LIBPATH and apps being ported would expect load time initialization which wouldn't work with this either.

Reply:
Ok, I'll remove zos-setlibpath.h from zoslib.

Points 1, 2 and 3 are clear, but curious how does one "enable the OS to save LIBPATH & STEPLIB into the executable"?

Reply:
We added something similar for perl and other projects using "environment variable hooks". Here's an example: https://github.com/ZOSOpenTools/perlport/blob/main/buildenv#L25. `zopen build` generates an .o file with function zoslib_env_hook (this replaces PROJECT_ROOT with the project's root directory): https://github.com/ZOSOpenTools/meta/blob/c7c98b0a431da1d9c55df68aca447ecf1a37a83c/bin/zopen-build#L2021 and zoslib checks for its presense here: https://github.com/ibmruntimes/zoslib/blob/zopen/src/zos.cc#L2689

Reply:
Ok, thanks Igor. I was referring to the case in which an app uses zoslib outside the zopen environment, like Node.js. I've now deleted zos-setlibpath.h.

Reply:
>"enable the OS to save LIBPATH & STEPLIB into the executable"
This means providing a way to embed a default LIBPATH/STEPLIB into the executable.  On aix, all of the -L paths or the linker option `-blibpath` are saved in the header section.  This can be seen with `dump -Hv`.  Elf has a similar set up.  The -R link option adds the paths.